### PR TITLE
add Auxiliary.bpcon and Auxiliary.dscon

### PR DESCRIPTION
**Pull request by mercury233**
There are over 200 scripts repeating those codes

### CODE REVIEW

**Comment by DailyShana** [comment]
please use intuitive names

**Comment by mercury233** [comment]
it is annoying to write long function names

**Comment by DailyShana** [comment]
functions in `aux` are just like standard library in c++, they are part of the script api

**Comment by mercury233** [comment]
we had many short function names and variable names already, so for keeping API style we should use more in future

File: utility.lua [hunk 1]
```diff
@@ -2090,6 +2090,14 @@ end
 function Auxiliary.exccon(e)
 	return Duel.GetTurnCount()~=e:GetHandler():GetTurnID() or e:GetHandler():IsReason(REASON_RETURN)
 end
+--condition of checking battle phase availability
+function Auxiliary.bpcon(e,tp,eg,ep,ev,re,r,rp)
+	return Duel.IsAbleToEnterBP() or (Duel.GetCurrentPhase()>=PHASE_BATTLE_START and Duel.GetCurrentPhase()<=PHASE_BATTLE)
```
Comment (DailyShana) [inline utility.lua]: if `Duel.GetCurrentPhase()>=PHASE_BATTLE_START and Duel.GetCurrentPhase()<=PHASE_BATTLE` is always needed, we should consider merge is condition into `Duel.IsAbleToEnterBP`

**Pull request merged**